source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -1230,7 +1235,9 @@ class IstioStatsFilter : public Http::PassThroughFilter,
                                                                ? context_.unknown_
                                                                : pool_.add(service_host_name)});
       tags_.push_back({context_.destination_service_namespace_,
-                       !peer_namespace.empty() ? pool_.add(peer_namespace) : context_.unknown_});
+                       !service_namespace.empty()
+                           ? pool_.add(service_namespace)
+                           : (peer ? pool_.add(peer->namespace_name_) : context_.unknown_)});
```

Comment:
why change this from `!peer_namespace.empty() ? pool_.add(peer_namespace) : context_.unknown_`?

Reply:
We were relying exclusively on endpoint metadata for destination_service_namespace. This makes sense for workload based metrics, but for service metrics, it has the unfortunate side effect of losing information when there are no healthy backends. So instead, we petard with using cluster metadata if it exists, falling back to endpoint metadata after that

Reply:
just L1240, why not use prev expression?

Reply:
Ah I see what you mean; I'll switch it around so it's consistent